iroh-bytes/examples/provide-bytes.rs

```diff
@@ -0,0 +1,119 @@
+//! An example that provides a collection over a Quinn connection.
+//!
+//! Since this example does not use `iroh-net::MagicEndpoint`, it does not do any holepunching, and so will only work locally or between two processes that have public IP addresses.
+//!
+//! Run this example with
+//!    cargo run --example provide-bytes blob
+//! To provide a blob (single file)
+//!
+//! Run this example with
+//!    cargo run --example provide-bytes collection
+//! To provide a collection (multiple blobs)
+use anyhow::Result;
+use tokio_util::task::LocalPoolHandle;
+use tracing_subscriber::{prelude::*, EnvFilter};
+
+use iroh_bytes::{format::collection::Collection, Hash};
+
+mod connect;
+use connect::{make_and_write_certs, make_server_endpoint, CERT_PATH};
+
+// set the RUST_LOG env var to one of {debug,info,warn} to see logging info
+pub fn setup_logging() {
+    tracing_subscriber::registry()
+        .with(tracing_subscriber::fmt::layer().with_writer(std::io::stderr))
+        .with(EnvFilter::from_default_env())
+        .try_init()
+        .ok();
+}
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let args: Vec<_> = std::env::args().collect();
+    if args.len() != 2 {
+        anyhow::bail!(
+            "usage: provide-bytes [FORMAT], where [FORMAT] is either 'blob' or 'collection'\n\nThe 'blob' example demonstrates sending a single blob of bytes. The 'collection' example demonstrates sending multiple blobs of bytes, grouped together in a 'collection'."
+        );
+    }
+    let format = {
+        if args[1] != "blob" && args[1] != "collection" {
+            anyhow::bail!(
+                "expected either 'blob' or 'collection' for FORMAT argument, got {}",
+                args[1]
+            );
+        }
+        args[1].clone()
+    };
```

Comment:
I split "blob" and "collection" up to make the instructions easier for the user when running the example, but I'm not sure this is the best way to do it.

Could possibly split this into `provide-blob` and `provide-collection`.

Reply:
I think it is fine.